app/src/main/java/com/gholem/moneylab/features/splashScreen/viewmodel/SplashViewModel.kt

```diff
@@ -29,8 +30,11 @@ class SplashViewModel @Inject constructor(
 
     var navigation: NavigationLiveData<SplashNavigationEvent> = NavigationLiveData()
 
-    fun getCategories() = viewModelScope.launch {
-        checkIfCategoriesIsEmpty()
+    private var listOfCategories = mutableListOf<TransactionCategory>()
```

Comment:
You don't need this variable. Just make single method where you fetch categories from database and if the result is empty then save the default list of categories